tests/integration/local/invoke/invoke_cdk_templates.py

```diff
@@ -0,0 +1,59 @@
+import json
+import docker
+from docker.errors import APIError
+from parameterized import parameterized
+
+from samcli import __version__ as version
+from samcli.local.docker.lambda_image import RAPID_IMAGE_TAG_PREFIX
+from tests.integration.local.invoke.invoke_integ_base import InvokeIntegBase
+from samcli.lib.utils.architecture import X86_64
+
+
+class TestCDKSynthesizedTemplatesFunctions(InvokeIntegBase):
+
+    template = "cdk/cdk_template.yaml"
+    functions = [
+        "StandardFunctionConstructZipFunction",
+        "StandardFunctionConstructImageFunction",
+        "DockerImageFunctionConstruct",
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        # Run sam build first to build the image functions
+        super(TestCDKSynthesizedTemplatesFunctions, cls).setUpClass()
+        build_command_list = super().get_build_command_list(cls, template_path=cls.template_path)
+        super().run_command(cls, command_list=build_command_list)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        # Remove images
+        docker_client = docker.from_env()
```

Comment:
This feels like we could abstract this into the base class but maybe there is a reason that won't work.

Why do this on `tearDownClass` over `teardown`? I expected to tear down after each function but maybe this is an optimization on tests? Whichever the reason, can you document it for clarity. Just helps us understand the thoughts behind things in the future.

Reply:
This logic is inherent to this class. Not all classes that extend the base class need to be built and cleaned up. 

As for `tearDownClass` over `teardown`, we only need to run it once when we finish with these Docker images. Re-building and re-removing the images after each individual test will eat up a lot of unnecessary time since they don't change between tests.

Reply:
That was my guess but would be good to just document that inline. Might be good for us to apply this logic across our tests. Flagging to @mndeveci and @CoshUS who I think are looking at some improvements in this area. Might not work in all cases but could be general useful for some things at least.